crates/primitives/src/transaction/mod.rs

```diff
@@ -502,31 +595,38 @@ impl From<TxEip4844> for Transaction {
 }
 
 impl Compact for Transaction {
+    // Serializes the TxType to the buffer if necessary, returning 2 bits of the type as an
+    // identifier instead of the length.
     fn to_compact<B>(self, buf: &mut B) -> usize
     where
         B: bytes::BufMut + AsMut<[u8]>,
     {
+        let identifier = self.tx_type().to_compact(buf);
         match self {
             Transaction::Legacy(tx) => {
                 tx.to_compact(buf);
-                0
             }
             Transaction::Eip2930(tx) => {
                 tx.to_compact(buf);
-                1
             }
             Transaction::Eip1559(tx) => {
                 tx.to_compact(buf);
-                2
             }
             Transaction::Eip4844(tx) => {
                 tx.to_compact(buf);
-                3
+            }
+            #[cfg(feature = "optimism")]
+            Transaction::Deposit(deposit) => {
+                deposit.to_compact(buf);
```

Comment:
nit consistency `s/deposit/tx`

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/161